### ISSUE TITLE
Crate layout of Unicode Properties

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#1136]
As of https://github.com/unicode-org/icu4x/pull/1114, we have:

- UnicodeSet in `components/uniset`
- Unicode properties in `components/uniset/src/props.rs`
- CodePointTrie in `utils/codepointtrie`

I think we should move `components/uniset` to `utils/uniset`, but we should keep the properties stuff in a new crate such as `components/properties` or `components/uprops`.

Thoughts?  Approval?

- [ ] @echeran 
- [ ] @iainireland 
- [ ] @zbraniecki

**Comment by iainireland** [comment]
Yeah, I've been thinking we should do something along these lines for a while.

**Comment by echeran** [comment]
Is a/the guiding principle here that `utils` includes data structures like `FixedDecimal`, `CodePointTrie`, `UnicodeSet`, and the user-facing APIs live in `components`? If so, everything above makes sense.
